### ISSUE TITLE
Simple call to User funciton errors

### ISSUE THREAD

**Issue by TomMiller-mas** [kibertoad/toad-scheduler#104]
I am trying to call a function to initiate an email process function:
I would like to use async/await, but it seems to want also sorts of additional stuff that I don't need.
This is the error:  **Error while handling task Process Email: procEmail is not defined**

```
module.exports.scheduleEmail = function scheduleEmail() {
  dlog("Trace", "scheduleEmail");
  const scheduler = new ToadScheduler();

  const task = new Task("Process Email", () => {
    procEmail();
  });

  const job_01 = new SimpleIntervalJob({ seconds: 20 }, task, "id_01");

  scheduler.addSimpleIntervalJob(job_01);
};

module.exports.procEmail = function procEmail() {
  dlog("Trace", "Enter procEmail");
};
```

**Comment by TomMiller-mas** [comment]
I figured out my error about the same time you responded.  Now I would like to get it working with async/await.  Please see my second block of code because it is not working. I am fairly new to javascript, so I really appreciate the help!

This works:
```
module.exports.scheduleEmail = function scheduleEmail() {
  dlog("Trace", "scheduleEmail");
  const scheduler = new ToadScheduler();

  const task = new Task("Process Email", () => {
    procEmail();
  });

  const job_01 = new SimpleIntervalJob({ seconds: 30 }, task, "id_01");

  scheduler.addSimpleIntervalJob(job_01);
};

function procEmail() {
  dlog("Trace", "Enter procEmail");
}

module.exports.procEmail = procEmail;
```
This is not working:
```
module.exports.scheduleEmail = async function scheduleEmail() {
  dlog("Trace", "scheduleEmail");
  const scheduler = new ToadScheduler();

  const task = new AsyncTask("Process Email", () => {
    try {
      procEmail();
    } catch {}
  });

  const job_01 = new SimpleIntervalJob({ seconds: 30 }, task, "id_01");

  scheduler.addSimpleIntervalJob(job_01);
};

async function procEmail() {
  try {
    dlog("Trace", "Enter procEmail");
  } catch {}
}

module.exports.procEmail = procEmail;
```

```
Trace: Enter procEmail
C:\devSandbox-Tom\DevFork-Tom\MAS-EMRESTServer_v98\mas-rest-srv\node_modules\toad-scheduler\dist\lib\common\AsyncTask.js:13
        this.handler().catch((err) => {
                      ^

TypeError: Cannot read properties of undefined (reading 'catch')
    at AsyncTask.execute (C:\devSandbox-Tom\DevFork-Tom\MAS-EMRESTServer_v98\mas-rest-srv\node_modules\toad-scheduler\dist\lib\common\AsyncTask.js:13:23)
    at Timeout._onTimeout (C:\devSandbox-Tom\DevFork-Tom\MAS-EMRESTServer_v98\mas-rest-srv\node_modules\toad-scheduler\dist\lib\engines\simple-interval\SimpleIntervalJob.js:26:23)
    at listOnTimeout (node:internal/timers:559:17)
    at processTimers (node:internal/timers:502:7)
[nodemon] app crashed - waiting for file changes before starting...
```

**Comment by TomMiller-mas** [comment]
OK, that worked.  Being new to javascript, why "return" vs "await"?

**Comment by kibertoad** [comment]
await should have worked too, I think, you would still implicitly return promise then. current implementation returns nothing, hence crashes
